ui/apps/platform/src/Containers/ComplianceEnhanced/ScanConfigs/compliance.scanConfigs.utils.ts

```diff
@@ -0,0 +1,120 @@
+import {
+    DailySchedule,
+    MonthlySchedule,
+    Schedule,
+    ScheduleBase,
+    UnsetSchedule,
+    WeeklySchedule,
+} from 'services/ComplianceEnhancedService';
+import { getDayOfMonthWithOrdinal, getTimeHoursMinutes } from 'utils/dateUtils';
+
+import { ScanConfigFormValues, ScanConfigParameters } from './Wizard/useFormikScanConfig';
+
+export function convertFormikParametersToSchedule(parameters: ScanConfigParameters): Schedule {
+    const { intervalType, time, daysOfWeek, daysOfMonth } = parameters;
+
+    // Convert the time to hour and minute
+    const [hourString, minuteString] = time.split(/[: ]+/);
+    let hour = parseInt(hourString, 10);
+    const minute = parseInt(minuteString, 10);
+
+    // Convert 12-hour format to 24-hour format
+    if (time.includes('PM') && hour < 12) {
+        hour += 12;
+    }
+    if (time.includes('AM') && hour === 12) {
+        hour = 0;
+    }
+
+    const baseSchedule: ScheduleBase = {
+        hour,
+        minute,
+    };
+
+    switch (intervalType) {
+        case 'WEEKLY': {
+            const weeklySchedule: WeeklySchedule = {
+                ...baseSchedule,
+                intervalType: 'WEEKLY',
+                daysOfWeek: { days: daysOfWeek.map((day) => parseInt(day)) },
+            };
+            return weeklySchedule;
+        }
+
+        case 'MONTHLY': {
+            const monthlySchedule: MonthlySchedule = {
+                ...baseSchedule,
+                intervalType: 'MONTHLY',
+                daysOfMonth: { days: daysOfMonth.map((day) => parseInt(day)) },
+            };
+            return monthlySchedule;
+        }
+
+        case 'DAILY': {
+            const dailySchedule: DailySchedule = {
+                ...baseSchedule,
+                intervalType: 'DAILY',
+            };
+            return dailySchedule;
+        }
+
+        case null:
+        default: {
+            const unsetSchedule: UnsetSchedule = {
+                ...baseSchedule,
+                intervalType: 'UNSET',
+            };
+            return unsetSchedule;
+        }
+    }
+}
+
+export function convertFormikToScanConfig(formikValues: ScanConfigFormValues) {
```

Comment:
Is there a function return type that would help TypeScript double-check the literal object below?